tests/mesh/test-hybrid-solid.cpp

```diff
@@ -0,0 +1,273 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+#include <geode/basic/attribute_manager.h>
+#include <geode/basic/logger.h>
+
+#include <geode/geometry/bounding_box.h>
+#include <geode/geometry/point.h>
+#include <geode/geometry/vector.h>
+
+#include <geode/mesh/builder/geode_hybrid_solid_builder.h>
+#include <geode/mesh/builder/solid_edges_builder.h>
+#include <geode/mesh/builder/solid_facets_builder.h>
+#include <geode/mesh/core/geode_hybrid_solid.h>
+#include <geode/mesh/core/solid_edges.h>
+#include <geode/mesh/core/solid_facets.h>
+#include <geode/mesh/io/hybrid_solid_input.h>
+#include <geode/mesh/io/hybrid_solid_output.h>
+
+#include <geode/tests/common.h>
+
+void test_create_vertices( const geode::HybridSolid3D& hybrid_solid,
+    geode::HybridSolidBuilder3D& builder )
+{
+    builder.create_point( { { 0, 0, 0 } } );
+    builder.create_point( { { 1, 0, 0 } } );
+    builder.create_point( { { 2, 1, 0 } } );
+    builder.create_point( { { 1, 2, 0 } } );
+    builder.create_point( { { 0, 2, 0 } } );
+    builder.create_point( { { 0, 0, 1 } } );
+    builder.create_point( { { 1, 0, 1 } } );
+    builder.create_point( { { 2, 1, 1 } } );
+    builder.create_point( { { 1, 2, 1 } } );
+    builder.create_point( { { 0, 2, 1 } } );
+    builder.create_point( { { 1, 1, 2 } } );
+    OPENGEODE_EXCEPTION( hybrid_solid.isolated_vertex( 0 ),
+        "[Test] Vertices should be isolated before polyhedra creation" );
+    OPENGEODE_EXCEPTION( hybrid_solid.nb_vertices() == 11,
+        "[Test] HybridSolid should have 8 vertices" );
+}
+
+void test_bounding_box( const geode::HybridSolid3D& hybrid_solid )
+{
+    geode::Point3D answer_min{ { 0, 0, 0 } };
+    geode::Point3D answer_max{ { 2, 2, 2 } };
+    const auto bbox = hybrid_solid.bounding_box();
+    OPENGEODE_EXCEPTION( bbox.min() == answer_min,
+        "[Test] Wrong computation of bounding box (min)" );
+    OPENGEODE_EXCEPTION( bbox.max() == answer_max,
+        "[Test] Wrong computation of bounding box (max)" );
+}
+
+void test_facets( const geode::HybridSolid3D& hybrid_solid )
+{
+    OPENGEODE_EXCEPTION(
+        hybrid_solid.facets().facet_from_vertices( { 0, 1, 3, 4 } ) == 0,
+        "[Test] Wrong facet from vertices" );
+}
```

Comment:
test_facetS testing only one facet. Is it enough? Maybe... until we show that was not.